tools/hsmtool.c

```diff
@@ -368,6 +372,148 @@ static int guess_to_remote(const char *address, struct node_id *node_id,
 	return 1;
 }
 
+static void get_words(struct words **words) {
+	printf("Select your language:\n");
+	printf("  0) English (en)\n");
+	printf("  1) Spanish (es)\n");
+	printf("  2) French (fr)\n");
+	printf("  3) Italian (it)\n");
+	printf("  4) Japanese (jp)\n");
+	printf("  5) Chinese simplified (zhs)\n");
+	printf("  6) Chinese traditional (zht)\n");
+	printf("Select [0-6]: ");
+
+	char *selected = NULL;
+	size_t size = 0;
+	size_t characters = getline(&selected, &size, stdin);
+	if (characters < 0)
+		errx(ERROR_USAGE, "Could not read line from stdin.");
+
+	int selection = selected[0] - '0';
+	if (characters != 2 || selection < 0 || selection > 6)
```

Comment:
`scanf` would be more appropriate here, rather than checking the length of the string and doing the char-to-int conversion.

Reply:
I changed it to `getline()` because `scanf` left the `\n` char and sequential `getline()`failed. What is the best way to deal with that?